app/validators/return-logs/setup/period-used.validator.js

```diff
@@ -55,10 +55,14 @@ function _fromDateBeforeToDate(value, helpers) {
 }
 
 function _fullDate(day, month, year) {
+  if (!year && !month && !day) {
+    return null
+  }
+
   const paddedMonth = month ? leftPadZeroes(month, 2) : ''
   const paddedDay = day ? leftPadZeroes(day, 2) : ''
 
-  return `${year}-${paddedMonth}-${paddedDay}`
+  return new Date(`${year}-${paddedMonth}-${paddedDay}`)
 }
```

Comment:
I assume that this `if` is to stop the `new Date` on line 65 from blowing up if any values are missing. As `new Date` will go pop if any of the values are missing, I'd use `OR` rather than `AND` in the if statement as all values are needed. Also on lines 62 & 63, if the month or day didn't exist and an empty string was used this would still cause the `new Date` to break. Apart from that, when creating the new date it is not required to pad the day and month with zeros.
If re-write the function like this:
```suggestion
function _fullDate(day, month, year) {
  if (!year || !month || !day) {
    return null
  }

  return new Date(`${year}-${month}-${day}`)
}
```